src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java

```diff
@@ -145,16 +147,36 @@ private void handleRotatedLogs(@Nonnull final String gcLogFileLocation, Containe
     @CheckForNull
     private String getGcLogFileLocation() {
 
-        String gcLogSwitch = vmArgumentFinder.findVmArgument(GCLOGS_JRE_SWITCH);
-        if (gcLogSwitch == null) {
-            LOGGER.fine("No GC Logging switch found, cannot collect gc logging files.");
-            return null;
+        String fileLocation;
+        
+        if(!isJava8OrBelow()) {
+            String gcLogSwitch = vmArgumentFinder.findVmArgument(GCLOGS_JRE9_SWITCH);
+            if (gcLogSwitch == null) {
+                LOGGER.fine("No GC Logging switch found, cannot collect gc logging files.");
+                fileLocation = null;
+            } else {
+                Matcher fileLocationMatcher = Pattern.compile(GCLOGS_JRE9_LOCATION).matcher(gcLogSwitch);
+                if(fileLocationMatcher.matches()) { 
+                    fileLocation = fileLocationMatcher.group("location");
+                } else {
+                    LOGGER.fine("No GC Logging custom file location found.");
+                    fileLocation = null;
+                }
+            }
+        } else {
+            String gcLogSwitch = vmArgumentFinder.findVmArgument(GCLOGS_JRE_SWITCH);
+            if (gcLogSwitch == null) {
+                LOGGER.fine("No GC Logging switch found, cannot collect gc logging files.");
+                fileLocation = null;
+            } else {
+                fileLocation = gcLogSwitch.substring(GCLOGS_JRE_SWITCH.length());
+            }
         }
-        return gcLogSwitch.substring(GCLOGS_JRE_SWITCH.length());
+        return fileLocation;
     }
     
     public boolean isFileLocationParameterized(String fileLocation) {
-        return Pattern.compile(".*%[tnp].*").matcher(fileLocation).find();
+        return Pattern.compile(".*%[tp].*").matcher(fileLocation).find();
```

Comment:
Is this related to the change?

Reply:
This was a leftover of a previous PR https://github.com/jenkinsci/support-core-plugin/pull/182/commits that I missed: https://github.com/jenkinsci/support-core-plugin/pull/182/commits/66970ac42f0f1e7223bad7c38c951446ec8dc68d. But did not affect the feature.